winpty/src/lib.rs

```diff
@@ -209,7 +209,7 @@ impl<'a, 'b> Winpty<'a> {
     /// Change the size of the Windows console window.
     /// 
     /// cols & rows MUST be greater than 0
-    pub fn set_size(&mut self, cols: usize, rows: usize) -> Result<(), Err> {
+    pub fn set_size(&mut self, cols: u16, rows: u16) -> Result<(), Err> {
```

Comment:
Why was this changed to a `u16`? Is the 'invalid value' you're talking about outside of the `u16` range and you're using the cast to `u16` to limit it to a smaller value? That seems like an odd approach, especially because it will just truncate bits and still accept the value if it's above `u16`, just that the value is now a `u16` which is completely unrelated to the original size.

Might just be a misunderstanding, but looking only at the changed code, this seems like an odd change.

Reply:
winpty accepts an `i32`, of which above zero values are valid (hence `u16` useful values).

Ideally there would be a saturating cast for `u32` -> `u16` but that doesn't exist so truncation seems appropriate given how unlikely it is to occur in the first place. It's mostly just to make sure we don't wrap around to `<0` like happened in the linked issue.

Reply:
I'm not a huge fan of this solution. It's not very obvious for developers unaware of the issue and it doesn't actually solve the problem, but just hides it.

Either we say it never happens and make sure it never does, or we find a way to properly handle the edge case and continue with correct behavior. I think that truncating the value and hoping that it will be fine, might come back to haunt us in the future, I'd strive for a long-term solution rather than the simplest fix to this specific problem.

Reply:
Having the bindings accept `u16` is the right decision I think, because it's the most appropriate type considering the reasonable values.

We could assert that the values that get passed in are less than `u16::MAX` which will make it more obvious if an underflow happens again.

Reply:
Nevermind, I just realised that a `u16` is not half of  a`u32`...

Reply:
Accepting `u16` is fine, I don't have a problem with putting the most fitting type restriction on the method, that's the purpose of these types after all.

My problem is indeed with just casting it to a `u16`. There already is a `> 0` assertion in the code, it might be possible to skip the `set_size` call when the value is out of range, right?

I'm not 100% up to date with the issue in question, but this might be the expected behavior, so invalid resize requests are ignored?

Reply:
The bindings make it a hard error because it's a hard error in winpty (also fails an assert there). The call is skipped in alacritty's `on_resize` method.

Reply:
Most of my comments here are more about the `on_resize` call rather than the method definition here.

I'd just handle a value above `u16::MAX` the same way that a value smaller or equal to `0` is handled.

I think it would probably be nice to get this fixed up for v0.2.2. I don't think the glutin PR is going to be merged in time, but since v0.2.2 is the first official Windows release, having fixed the basic issues with makes sense.

Reply:
I'm done with exams on the 23rd of November, may have to wait til around then before I can change it.

Reply:
No worries, this isn't super essential so I wouldn't hold back a minor version bump for it.

There's always a next release and with the glutin update waiting around, chances are we'll be able to bundle it with that.